test/v5/kyberBancorReserve.js

```diff
@@ -0,0 +1,576 @@
+        let amountBnt = (new BigNumber(10)).pow(18).mul(500);
+        let ethToBntRate = await reserve.getConversionRate(ethAddress, bancorBntToken.address, amountEth, 0);
+        let bntToEthRate = await reserve.getConversionRate(bancorBntToken.address, ethAddress, amountBnt, 0);
+
+        assert.notEqual(ethToBntRate.valueOf(), 0, "should have rate eth to bnt");
+        assert.notEqual(bntToEthRate.valueOf(), 0, "should have rate bnt to eth");
+
+        await reserve.disableTrade({from: alerter});
+        try {
+            await reserve.trade(ethAddress, amountEth, bancorBntToken.address, user, ethToBntRate, true, {from: network, value: amountEth});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+
+        await bancorBntToken.transfer(network, amountBnt);
+        try {
+            await reserve.trade(bancorBntToken.address, amountBnt, ethAddress, user, bntToEthRate, true, {from: network});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+
+        await reserve.enableTrade({from: admin});
+    });
+
+    it("Should test trade msg sender is not network", async function() {
+        let amountEth = (new BigNumber(10)).pow(18);
+        let amountBnt = (new BigNumber(10)).pow(18).mul(500);
+        let ethToBntRate = await reserve.getConversionRate(ethAddress, bancorBntToken.address, amountEth, 0);
+        let bntToEthRate = await reserve.getConversionRate(bancorBntToken.address, ethAddress, amountBnt, 0);
+
+        try {
+            await reserve.trade(ethAddress, amountEth, bancorBntToken.address, user, ethToBntRate, true, {from: user, value: amountEth});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+
+        try {
+            await reserve.trade(bancorBntToken.address, amountBnt, ethAddress, user, bntToEthRate, true, {from: user});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+    });
+
+    it("Should test trade token is invalid (src & dest are not ETH or token is not bnt)", async function() {
+        let amountEth = (new BigNumber(10)).pow(18);
+        let amountBnt = (new BigNumber(10)).pow(18).mul(500);
+        let ethToBntRate = await reserve.getConversionRate(ethAddress, bancorBntToken.address, amountEth, 0);
+        let bntToEthRate = await reserve.getConversionRate(bancorBntToken.address, ethAddress, amountBnt, 0);
+
+        let token = await TestToken.new("test token", "tst", 18);
+        await token.transfer(user, (new BigNumber(2)).pow(18).mul(1000));
```

Comment:
should transfer network.
No?

Reply:
Oh yes should transfer to network, fixed